server/pragma-checker/index.js

```diff
@@ -0,0 +1,67 @@
+/***** WARNING: ES5 code only here. Not transpiled! *****/
+
+/**
+ * External dependecies
+ */
+var error = require( 'chalk' ).bold.red;
+var info = require( 'chalk' ).bold.yellow;
+var startsWith = require( 'lodash/string/startsWith' );
+
+var PLUGIN_TITLE = 'PragmaChecker';
+var SSR_READY = '/** @ssr-ready **/';
+
+function PragmaCheckPlugin( options ) {
+	this.options = options || {};
+}
+
+function scanDependencies( module, compilation ) {
+	if ( ! module.dependencies ) {
+		return;
+	}
+
+	module.dependencies.forEach( function( dep ) {
+		// If the module is compiled through babel, we can be pretty sure it's our own module, not from npm.
+		if ( dep.module && /babel-loader/.test( dep.module.request ) &&
+				dep.module._source &&
+				dep.module._source._value.indexOf( SSR_READY ) === -1 ) {
+			compilation.errors.push( PLUGIN_TITLE + ': ' + module.rawRequest + ', dependency ' + dep.module.rawRequest + ' is not ' + SSR_READY );
+		}
+
+		if ( dep.module && dep.module.dependencies ) {
+			scanDependencies( dep.module.dependencies );
+		}
```

Comment:
Since both `if` blocks check for `dep.module &&`, you could just `continue` at the top of the loop `if ( ! dep.module )`...